### ISSUE TITLE
[Improvement] Page is loading every time even if there is no change in any component

### ISSUE THREAD

**Issue by Vishwa-Desai** [rohitdasu/projectmate#448]
### Description

I am not sure if its bug or not but when i was roaming around your website i felt that on projects and mates page, it is unnecessary loading page every time when i switch to another tab and come back to you website even if there no change or new data to show.

**Comment by rohitdasu** [comment]
Hi @Vishwa-Desai Thanks for reporting the issue. We'll check it out 👍

**Comment by csagar131** [comment]
Hi @Vishwa-Desai  this is not a bug, this is because project is using SWR(stale-while-revalidate) library for data fetching and caching, so when you switch the tab and come back later, it ensures that you will get the latest and sync data for that its making an api request to fetch the fresh data when you come back...
refer this for more info : https://swr.vercel.app/docs/revalidation

**Comment by rohitdasu** [comment]
Yup that's the cause of the reload. Thanks @csagar131 for checking it 👍

**Comment by rohitdasu** [comment]
@Vishwa-Desai i hope you got the reason for the issue. We cannot remove the revalidation feature as it serves users with the latest data, but we can improve the UX by not showing the loader(which causes a flicker) when it revalidates the data. So i am replacing the bug tag from this issue with "Improvement" tag. it's open for anyone to work on this issue.

**Comment by Vishwa-Desai** [comment]
@rohitdasu Yes i got it, Thanks for looking into it.

**Comment by Delving-Mage** [comment]
What's the status of this? @rohitdasu , if it's open can i work on this?

**Comment by ravijha9546** [comment]
@rohitdasu Hi. Is this still open ? I would like to work on it.
